open-telemetry/opentelemetry-erlang
# Generate docs with ExDoc

This PR add ability to generate documentation for `opentelemetry` and `opentelemetry_exporter` with ExDoc. This work is mostly refer to the following resources: 

- [ExDoc README](https://github.com/elixir-lang/ex_doc#using-exdoc-with-erlang-projects)
- [beam-telemetry/telemetry](https://github.com/beam-telemetry/telemetry)

So far, there are a few caveats:

- I haven't thoroughly check and compare the generated docs for each modules and functions with the current docs.
- Not sure if the changes will work for publishing to Hex.pm and what are the workflow like. ExDoc README linked above does document the manual steps required.
- The `$version` used in `docs.sh` is hardcoded and used by both `opentelemetry` and `opentelemetry_exporter`.

Since I'm quite new to Erlang and Rebar, so there are probably places where I did things incorrectly or can be further improve.

**To test out locally:**

- Make sure you're running on OTP 24+
- Install ExDoc with `mix escript.install github elixir-lang/ex_doc
- Run `./docs.sh`.
- View the generated HTML by opening:
  - `apps/opentelemetry/doc/index.html`
  - `apps/opentelemetry_exporter/doc/index.html`
  
### Screenshot
  
#### OpenTelemetry
  
![Screenshot 2021-08-21 at 6 19 54 PM](https://user-images.githubusercontent.com/12683067/130318718-e96de344-3ba6-4e11-b133-5c934905ff08.png)
![Screenshot 2021-08-21 at 6 19 59 PM](https://user-images.githubusercontent.com/12683067/130318721-9d81cf0c-740c-4e17-a911-a0e5f1a771e8.png)
![Screenshot 2021-08-21 at 6 20 04 PM](https://user-images.githubusercontent.com/12683067/130318724-669fab80-46bc-42eb-8b13-89313ce41bb3.png)



 #### OpenTelemetry Exporter
  
![Screenshot 2021-08-21 at 6 20 34 PM](https://user-images.githubusercontent.com/12683067/130318737-d4b78fa0-820f-4417-a781-519c6d57e5ca.png)
![Screenshot 2021-08-21 at 6 20 39 PM](https://user-images.githubusercontent.com/12683067/130318741-dd34146b-933c-4a80-b206-dc5cbc42062b.png)
